packages/edit-post/src/components/sidebar/post-link/index.js

```diff
@@ -104,8 +107,18 @@ function PostLink( {
 			<div className="edit-post-post-link__preview-link-container">
 				<ExternalLink
 					className="edit-post-post-link__link"
-					href={ postLink }
+					href={ isEditable ? permalink : postLink }
 					target="_blank"
+					onClick={ ( event ) => {
+						// This shows the permalink in the status bar,
+						// while uses the postLink to open in a new window.
+						event.preventDefault();
+						window.open(
+							postLink,
+							'_blank',
+							'noreferrer,noopener'
+						);
+					} }
```

Comment:
I'd rather not hack the `<a href="" />`, since it makes for some really unexpected behavior. As others have elaborated in #41948, the permalink should actually work for drafts as well.

```suggestion
```